src/data/mystery-encounters/sleeping-snorlax.ts

```diff
@@ -0,0 +1,127 @@
+import BattleScene from "../../battle-scene";
+import {
+  EnemyPartyConfig,
+  initBattleWithEnemyConfig,
+  leaveEncounterWithoutBattle,
+  pushDialogueTokensFromPokemon,
+  setCustomEncounterRewards,
+  showEncounterText
+} from "./mystery-encounter-utils";
+import MysteryEncounter, {MysteryEncounterBuilder} from "../mystery-encounter";
+import * as Utils from "../../utils";
+import {MysteryEncounterType} from "#enums/mystery-encounter-type";
+import {MoveRequirement, StatusEffectRequirement, WaveCountRequirement} from "../mystery-encounter-requirements";
+import {MysteryEncounterOptionBuilder} from "../mystery-encounter-option";
+import {
+  modifierTypes
+} from "#app/modifier/modifier-type";
+import PokemonSpecies, { PokemonForm, SpeciesFormKey } from "../pokemon-species";
+import { Type } from "../type";
+import { Abilities } from "#enums/abilities";
+import { Species } from "#enums/species";
+import { GrowthRate } from "../exp";
+import { StatusEffect } from "../status-effect";
+import { Moves } from "#enums/moves";
+import { SummonPhase } from "#app/phases";
+
+export const SleepingSnorlaxEncounter: MysteryEncounter = new MysteryEncounterBuilder()
+  .withEncounterType(MysteryEncounterType.SLEEPING_SNORLAX)
+  .withIntroSpriteConfigs([]) // Set in onInit()
+  .withSceneRequirement(new WaveCountRequirement([10, 180])) // waves 10 to 180
+  .withCatchAllowed(true)
+  .withOnInit((scene: BattleScene) => {
+    const instance = scene.currentBattle.mysteryEncounter;
+    pushDialogueTokensFromPokemon(instance);
+    console.log(instance);
+    const availablePartyMembers = scene.getParty().filter(p => p.isAllowedInBattle());
+    if (!availablePartyMembers[0].isOnField()) {
+
+      scene.pushPhase(new SummonPhase(scene, 0));
+    }
+
+    // Calculate boss mon
+    const bossSpecies = new PokemonSpecies(Species.SNORLAX, 1, false, false, false, "Sleeping Pokémon", Type.NORMAL, null, 2.1, 460, Abilities.IMMUNITY, Abilities.THICK_FAT, Abilities.GLUTTONY, 540, 160, 110, 65, 65, 110, 30, 25, 50, 189, GrowthRate.SLOW, 87.5, false, true,
```

Comment:
Should be able to just get the instance of the species you want here unless you’re trying to make a totally custom snorlax species for some reason. I forget the exact name of the function but I believe it’s “getPokemonSpecies()” or something

Reply:
Example in another encounter: https://github.com/InnocentGameDev/PokeRogue-Events/blob/b4a63021234a0a035c4da92759f4cfa24838a14c/src/data/mystery-encounters/dark-deal.ts#L161

Reply:
nice, didn't know about that. @InnocentGameDev did mention maybe pumping up the snorlax, but i figure we can leave balancing until after